File: frontend/src/types/brand.ts
export interface BrandTarget {
  id: string;
  agency_id: string;
  creator_id: string;
  company_name: string;
  company_domain: string | null;
  reasoning: string | null;
  apollo_organization_id: string | null;
  status: "suggested" | "approved" | "blacklisted";
  scouting_status: string | null;
  discovery_round: number;
  created_at: string;
  updated_at: string;
}

export interface BrandDiscoveryStatus {
  status: string;
  discovery_round: number | null;
  brands_found: number | null;
}

File: backend/app/schemas/brand.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


# --- LLM output schemas ---

class BrandSuggestion(BaseModel):
    company_name: str
    company_domain: str | None = None
    reasoning: str
    fit_score: int
    suggested_collab_type: str


class BrandDiscoveryOutput(BaseModel):
    suggestions: list[BrandSuggestion]


# --- API request/response schemas ---

class BrandTargetResponse(BaseModel):
    id: UUID
    agency_id: UUID
    creator_id: UUID
    company_name: str
    company_domain: str | None
    reasoning: str | None
    apollo_organization_id: str | None
    status: str
    scouting_status: str | None
    discovery_round: int
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class BrandDiscoveryStatusResponse(BaseModel):
    status: str  # pending | processing | completed | failed
    discovery_round: int | None = None
    brands_found: int | None = None
